operator/poolboy_templating.py

```diff
@@ -154,17 +156,43 @@ def jinja2process(template, template_style='jinja2', variables={}):
         return template_out
 
 def recursive_process_template_strings(template, template_style='jinja2', variables={}):
+    omit = '__omit_place_holder__' + ''.join(random.choices('abcdef0123456789', k=40))
+    return __recursive_strip_omit(
+        __recursive_process_template_strings(
+            omit = omit,
+            template = template,
+            template_style = template_style,
+            variables = variables,
+        ),
+        omit = omit,
+    )
+
+def __recursive_process_template_strings(template, omit, template_style, variables):
     if isinstance(template, dict):
         return {
-            k: recursive_process_template_strings(v, template_style=template_style, variables=variables)
-            for k, v in template.items()
+            key: __recursive_process_template_strings(val, omit=omit, template_style=template_style, variables=variables)
+            for key, val in template.items()
         }
     elif isinstance(template, list):
         return [
-            recursive_process_template_strings(item, template_style=template_style, variables=variables)
+            __recursive_process_template_strings(item, omit=omit, template_style=template_style, variables=variables)
             for item in template
         ]
     elif isinstance(template, str):
-        return jinja2process(template=template, template_style=template_style, variables=variables)
+        return jinja2process(omit=omit, template=template, template_style=template_style, variables=variables)
```

Comment:
even if they are named parameter, i would add the new one at the end

Reply:
I prefer having named parameters alphabetized.

Reply:
Ah ok.

It's just that generally speaking, if anything uses it without the names, it can break.

```python
def bla(b=1, c=2, d=3):
    print(b, c, d)


b = 1
c = 2
d = 3
bla(b, c, d)   # prints 1 2 3
```

if you introduce `a`:

```python

def bla(a=0, b=1, c=2, d=3):
    print(b, c, d)

b = 1
c = 2
d = 3
bla(b, c, d) # prints 2 3 3
```

Reply:
I realized i didn't highlight the right place:

```diff
- def jinja2process(template, template_style='jinja2', variables={}):
+ def jinja2process(template, omit=None, template_style='jinja2', variables={}):
```

Reply:
Okay, changing the call to match the arguments and pass template as positional.